### PR TITLE
prepping for Sensu Go.

### PR DESCRIPTION
**Pull request by jspaleta**
## Pull Request Checklist

**Is this in reference to an existing issue?**
closes:   #41 #40 #39 #38 #42 
#### General

- [X] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [X] Update README with any necessary configuration snippets

- [ ]  Binstubs are created if needed

- [X] RuboCop passes

- [X] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose
1. Sensu Go Enablement
2. Update development deps to latest versions

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +34,7 @@ This CHANGELOG follows the format listed [here](https://github.com/sensu-plugins
 ### Fixed
 - check-log.rb: pattern is now required via option config (@cgeers)
 - check-log.rb: update descriptions of crit and warn options (@cgeers)
+>>>>>>> upstream/master
```
Comment (majormoses) [inline CHANGELOG.md]: looks like missed from a merge conflict resolution.

File: sensu-plugins-logs.gemspec [hunk 2]
```diff
@@ -30,21 +30,22 @@ Gem::Specification.new do |s| # rubocop:disable Metrics/BlockLength
   s.test_files             = s.files.grep(%r{^(test|spec|features)/})
   s.version                = SensuPluginsLogs::Version::VER_STRING
 
-  s.add_runtime_dependency 'sensu-plugin', '~> 2.7'
+  s.add_runtime_dependency 'sensu-plugin', '>= 3.0', '< 4.0'
```
Comment (majormoses) [inline sensu-plugins-logs.gemspec]: Thats fine but I'd prefer we keep the `~> 3.0` syntax to be consistent.

Comment (jspaleta) [reply sensu-plugins-logs.gemspec]: that was the suggestion for from the dependbot pr. I was rolling it in.  Can we adjust the dependbot rules to make the prefered suggestion?

File: CHANGELOG.md (hunk 1 above)
Comment (jspaleta) [reply CHANGELOG.md]: fixed in latest commit

File: sensu-plugins-logs.gemspec (hunk 2 above)
Comment (jspaleta) [reply sensu-plugins-logs.gemspec]: fixed in latest commit

Comment (majormoses) [reply sensu-plugins-logs.gemspec]: Ya I asked them about it they can in a `Gemfile` but not currently in a `gemspec` though its possible its just not an easy change to make apparently. We can with some exceptions. Sometimes dependabot will allow multiple versions of a major version which makes it possible to unintentionally pull in breaking changes. I am fine with `'>= 3.0', < 4.0` but I am not fine with `>= 2.0', < 4.0`.

**Comment by majormoses** [comment]
have you taken a look at the 2 failures, they seem legit, I will try to spend some time later looking at it and figuring out why it breaks on older than ruby 2.5 versions.

**Comment by jspaleta** [comment]
I think the failures are the result of the change i made to satisfy updated rubocup
changing  > 0  to .is_positive?     fails on older rubies.  Need to add an exception to rubocop

**Comment by jspaleta** [comment]
Build failures fixed with adjustment to rubocop rules.
Needed to disable Style/NumericPredicate  which was enforcing use of `positive?` to `replace > 0`
and not compatible with older rubies.

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,24 @@ This project adheres to [Semantic Versioning](http://semver.org/).
...
+- add ruby 2.5.1 into travis automation
+
+### Fixed
+- add rdoc to development dependancy to fix rake task errors ruby 2.5.1
+- update dev dep for github-markup
+- update dev dep for codeclimate-test-reporter 
+- update dev dep for serverspec
+- update dev dep for mixlib-shellout
+- update dev dep for rubocop
+- check-log.rb: pattern is now required via option config
```
Comment (majormoses) [inline CHANGELOG.md:23]: looks like this maybe a bad merge because the following two lines were released in 2.0.0 I can fix that up post merge.

**Review by majormoses — APPROVED** [review]
LGTM

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-logs/versions/3.0.0

**Pull request merged**